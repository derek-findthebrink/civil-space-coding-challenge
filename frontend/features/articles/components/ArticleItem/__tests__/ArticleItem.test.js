import * as React from "react";
import { render, cleanup, screen } from "@testing-library/react";

import { ArticleItem } from "..";

jest.mock('../../ArticleHeader', () => (
  {
    __esModule: true,
    ArticleHeader: () => (<h1>I am the ArticleHeader</h1>)
  }
))

afterEach(() => {
  cleanup()
})

describe("rendering", () => {
  let fragment
  let id
  let imageUrl
  let title
  let introduction
  let createdAt
  let author;

  beforeEach(() => {
    id = "123";
    imageUrl = 'https://www.image.com/image'
    title = 'Article Title'
    introduction = 'This is the introduction!'
    createdAt = '2020-08-16T14:30:00-08:00'
    author = {
      firstName: 'Todd',
      lastName: 'McGavin',
    }

    const { asFragment } = render(
      <ArticleItem
        id={id}
        imageUrl={imageUrl}
        title={title}
        introduction={introduction}
        createdAt={createdAt}
        author={author}
      />
    );
    fragment = asFragment();
  });

  it("matches snapshot", () => {
    expect(fragment).toMatchSnapshot();
  });

  it("renders an ArticleHeader", () => {
    expect(screen.getByRole('heading')).toHaveTextContent('I am the ArticleHeader');
  });

  it("renders the `introduction`", () => {
    expect(screen.getByText(introduction)).toBeTruthy();
  });

  it("renders the a `Read More` button", () => {
    expect(screen.getByRole('link')).toHaveTextContent('Read More');
  });
});
