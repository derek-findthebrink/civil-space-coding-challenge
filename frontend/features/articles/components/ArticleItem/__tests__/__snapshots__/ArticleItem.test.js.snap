// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering matches snapshot 1`] = `
<DocumentFragment>
  <article
    aria-label="Article"
    class="article-item"
    role="contentinfo"
  >
    <hr
      class="article-item__accent-hr"
    />
    <h1>
      I am the ArticleHeader
    </h1>
    <div
      class="sc-bdfBwQ HzSZM article-item__content-wrapper"
    >
      <div
        class="article-item__content"
      >
        <p
          class="article-item__intro"
        >
          This is the introduction!
        </p>
        <footer>
          <a
            class="article-item__read-more"
            href="/article/123"
            rel="noopener noreferrer"
            target="_blank"
          >
            Read More
          </a>
        </footer>
      </div>
    </div>
  </article>
</DocumentFragment>
`;
